modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/ImageMosaicReaderTest.java

```diff
@@ -3730,6 +3736,85 @@ public void testInitFromExistingStore() throws Exception {
         reader2.dispose();
     }
 
+    @Test
+    public void testMixedSampleModels() throws Exception {
+        File mosaicFolder = DataUtilities.urlToFile(mixedSampleModelURL);
+        cleanConfigurationFiles(mosaicFolder, mosaicFolder.getName());
+        final AbstractGridFormat format = TestUtils.getFormat(mixedSampleModelURL);
+        ImageMosaicReader reader = TestUtils.getReader(mixedSampleModelURL, format);
+
+        // Unsure about this test... if the mosaic is mixed, we will get a different
+        // color model depending on which area we hit, if it's a single granule, it
+        // will be that granule color model, and in general, it will depend on the combination
+        // ImageLayout il = reader.getImageLayout();
+        // assertThat(il.getSampleModel(null), instanceOf(ComponentSampleModel.class));
+        // assertThat(il.getSampleModel(null), instanceOf(ComponentColorModel.class));
+
```

Comment:
That comment contains a typo (not a big deal). It's checking whether a sampleModel is an instance of a colorModel :)